From c3e2d8ec2df477170080efe0b8089ec78a8347e1 Mon Sep 17 00:00:00 2001
From: Jason King <jason.king@joyent.com>
Date: Thu, 25 Oct 2018 21:25:54 +0000
Subject: [PATCH] OS-7328 lx recvmsg should allow cmsgs with no data Reviewed
 by: Patrick Mooney <patrick.mooney@joyent.com> Approved by: Dan McDonald
 <danmcd@joyent.com>

---
 usr/src/uts/common/brand/lx/syscall/lx_socket.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/usr/src/uts/common/brand/lx/syscall/lx_socket.c b/usr/src/uts/common/brand/lx/syscall/lx_socket.c
index 18bc92433d..e961abdac1 100644
--- a/usr/src/uts/common/brand/lx/syscall/lx_socket.c
+++ b/usr/src/uts/common/brand/lx/syscall/lx_socket.c
@@ -1009,7 +1009,7 @@ stol_cmsgs_copyout(void *input, socklen_t inlen, void *addr,
 		goto finish;
 	}
 
-	VERIFY(inlen > sizeof (struct cmsghdr));
+	VERIFY(inlen >= sizeof (struct cmsghdr));
 
 	/*
 	 * First determine how much space we need for the conversion and
-- 
2.21.0

